src/main/java/core/basesyntax/service/impl/TransactionProcessorImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.TransactionProcessor;
+import core.basesyntax.strategy.OperationHandlerFactory;
+import java.util.List;
+
+public class TransactionProcessorImpl implements TransactionProcessor {
+    private final OperationHandlerFactory handlerFactory;
+
+    public TransactionProcessorImpl(OperationHandlerFactory handlerFactory) {
+        this.handlerFactory = handlerFactory;
+    }
+
+    @Override
+    public void process(List<Transaction> transactions) {
+        for (Transaction transaction : transactions) {
+            handlerFactory.getHandler(transaction.getOperation())
+                    .handle(transaction.getFruit().getName(), transaction.getQuantity());
+        }
+    }
```

Comment:
Ensure that the `process` method is robust against possible failures in the operation handlers. If an exception occurs in one of the handlers, it could stop the processing of subsequent transactions. Consider adding error handling to continue processing the rest of the transactions even if one fails.